src/app/shared/model/image.model.ts

```diff
@@ -36,12 +34,10 @@ export class ImageModel extends UploadableModel {
   key() {
     if (this.doc) {
       return `prx.image.${this.doc.profileSubtype}.${this.doc.id}`;
-    } else if (this.parent) {
-      return `prx.image.new.${this.parent.profileType}.${this.parent.id}`;
-    } else if (this.grandparent) {
-      return `prx.image.draft.${this.grandparent.profileType}.${this.grandparent.id}`;
+    } else if (this.uuid) {
+      return `prx.image.new.${this.uuid}`;
```

Comment:
Much easier to use a uuid in the key, instead of relying on the parent/grandparent models.  Just requires that the parent model track which new-upload-uuids it has.